plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -1194,14 +1172,10 @@ static int set_log_group_retention(struct flb_cloudwatch *ctx, struct log_stream
 
         /* Check error */
         if (c->resp.payload_size > 0) {
-            error = flb_aws_error(c->resp.payload, c->resp.payload_size);
-            if (error != NULL) {
-                /* some other error occurred; notify user */
-                flb_aws_print_error(c->resp.payload, c->resp.payload_size,
-                                        "PutRetentionPolicy", ctx->ins);
-                flb_sds_destroy(error);
-            }
-            else {
+            /* some other error occurred; notify user */
+            is_error_parsed = flb_aws_print_error(c->resp.payload, c->resp.payload_size,
+                                               "PutRetentionPolicy", ctx->ins);
+            if (!is_error_parsed) {
```

Comment:
let's use fluent bit best practice of comparing directly with FLB_TRUE or FLB_FALSE. I think its more readable too.